### PR TITLE
Functions - College

### PR DESCRIPTION
**Pull request by Abduliitr**
## Checklist
- Solution is tested on sphere engine <br/>
    `Note:` Please copy only the solution part. Do not copy/paste the template code. That way, any issues with template code can be caught early on.
    - [x] Author
    - [x] Reviewer
- PR title is correct - [PR title conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#title-for-a-pull-request)
    - [x] Author
    - [x] Reviewer
- Folder/File naming is correct - [Naming conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#naming-of-files-and-folders)
    - [x] Author
    - [x] Reviewer
- Coding conventions are followed - [Coding conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#coding-conventions)
    - [x] Author
    - [x] Reviewer
- Test cases (action item for **reviewer**) <br/>
    - To be ensured by checking the problem on sphere-engine
    - If any test cases are missing, please add them - [how to add new test cases](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#adding-new-test-cases)
    - [x] There are at least 2 test cases
    - [x] There is at least 1 test case that is different from the sample test cases

### CODE REVIEW

**Comment by shivateja1729** [comment]
function name can be sumOfDivisors (camal case) as per the coding conventions.

**Comment by Abduliitr** [comment]
> function name can be sumOfDivisors (camal case) as per the coding conventions.

@LVSsandeepVarma please update this in the template..

At some places in the template, we have function names live `sample_function` , is that okay?

**Comment by LVSsandeepVarma** [comment]
@Abduliitr  updated the template

**Review by shivateja1729 — APPROVED** [review]

File: solutions/functions/college.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  return data[idx - 1];
+}
+
+// -------- Do NOT edit anything above this line ----------
+function sumofdivisors(n){
+    //Code here
+
+    let requiredSum = 0;
+
+    for(let i = 1; i <= (parseInt(Math.sqrt(n))+1); i++){
```
Comment (sailesh-ramayanam) [inline solutions/functions/college.js:17]: 1. Please use `Math.floor`
2. Move `sqrt(n) + 1` outside the `for` loop

File: solutions/functions/college.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        if(n % i === 0){
+            requiredSum += i;
+
+            if(n / i !== i){
+                requiredSum += n / i;
```
Comment (sailesh-ramayanam) [inline solutions/functions/college.js:22]: Please add a comment why we are doing this.

File: solutions/functions/college.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            }
+        }
+    }
+
+    return requiredSum;
+}
+    
+
+let n = parseInt(readLine())
+console.log(sumofdivisors(n))
```
Comment (sailesh-ramayanam) [inline solutions/functions/college.js:32]: Please add semi-colons (it is updated in the template).

**Review by sailesh-ramayanam — CHANGES_REQUESTED** [review]

File: solutions/functions/college.js (hunk 1 above)
Comment (Abduliitr) [reply solutions/functions/college.js:17]: Addressed.

File: solutions/functions/college.js (hunk 3 above)
Comment (Abduliitr) [reply solutions/functions/college.js:32]: Updated as per latest template.

File: solutions/functions/college.js (hunk 2 above)
Comment (Abduliitr) [reply solutions/functions/college.js:25]: Added.

**Review by sailesh-ramayanam — APPROVED** [review]

**Comment by shivateja1729** [comment]
> @shivateja1729 There doesn't seem to be a test case where `n` is a prime number. Please add it.

@sailesh-ramayanam  added a prime number as testcase.

**Pull request merged**